src/treq/client.py

```diff
@@ -147,14 +151,20 @@ def request(self, method, url, **kwargs):
         """
         method = method.encode('ascii').upper()
 
+        if isinstance(url, unicode):
+            parsed_url = URL.from_text(url)
+        else:
+            parsed_url = URL.from_text(url.decode('ascii'))
+
         # Join parameters provided in the URL
         # and the ones passed as argument.
         params = kwargs.get('params')
         if params:
-            url = _combine_query_params(url, params)
+            parsed_url = parsed_url.replace(
+                query=parsed_url.query + tuple(_coerced_query_params(params))
+            )
 
-        if isinstance(url, unicode):
-            url = URL.fromText(url).asURI().asText().encode('ascii')
+        url = bytes(parsed_url)
```

Comment:
Would you mind leaving the explicit `.as_uri().to_text().encode()` idiom here, rather than invoking it indirectly via `__bytes__` / `__str__`?  This works, but it's not documented super strictly in hyperlink and it's more or less a concession to `requests`' API design, not something I want to rely on really heavily.

Reply:
Sure, it looks like they're equivalent:

https://github.com/python-hyper/hyperlink/blob/56ed2291c0fc6a402b91db38d18d1ad4aed8efbf/src/hyperlink/_url.py#L1584-L1591

I'd hoped that `URL` might avoid some of those copies, but :man_shrugging:.